e3sm_diags/derivations/formulas_cosp.py

```diff
@@ -116,19 +116,20 @@ def cosp_bin_sum(target_var_key: str, var: xr.DataArray) -> xr.DataArray:
     # 4. Get tau range and lim.
     tau_range, tau_lim = _get_tau_subset_range_and_str(tau, tau_adj_range)
 
-    # 4. Get the axes mask conditional and subset the variable on it.
+    # 5. Get the axes mask conditional and subset the variable on it.
     cond = (
         (prs >= prs_range[0])
         & (prs <= prs_range[-1])
         & (tau >= tau_range[0])
         & (tau <= tau_range[-1])
     )
+
     var_sub = var.where(cond, drop=True)
 
-    # 5. Sum on axis=0 and axis=1 (tau and prs)
+    # 7. Sum on axis=0 and axis=1 (tau and prs)
     var_sum = var_sub.sum(dim=[prs.name, tau.name])
```

Comment:
RE: Issue 1 in https://github.com/E3SM-Project/e3sm_diags/pull/752#issuecomment-1945057995

This fixes missing outputs for the remaining variables: `ISCCPCOSP-CLDTOT_TAU1.3_9.4_ISCCP`, `ISCCPCOSP-CLDTOT_TAU1.3_ISCCP`, `MODISCOSP-CLDHGH/TOT_TAU1.3_9.4_MODIS`, `MODISCOSP-CLDHGH/TOT_TAU1.3_MODIS`

### Root Cause

1. The `prs_range` can be in descending order (e.g., `(90000.0, 9000.0)`) 
2. This caused the condition (`cond`) to not match any of the data values when subsetting with `.where()`
   - The condition being used is `>=90000.0 and <=9000.0` which is incorrect
   - The _correct_ condition should to be the other way around, `>= 9000 and <= 90000`
 3. Since there are no matching values, the below `numpy` error is thrown with `drop=True` (drop all `nan` values that don't match): 
   - `ValueError: Cannot apply_along_axis when any iteration dimensions are 0`

### Solution 

Sort `prs_range` and `tau_range` before subsetting

### Side-note

The CDMS2 version of sub-setting a variable is pretty confusing. Notice below how `prs_low` and `prs_high` are used to subset, but `prs_low` is actually the larger value. 

I think CDMS2 swaps both values around and subsets to include all values within the range. It uses the correct sub-setting condition, `>= 9000 and <= 90000`. 

https://github.com/E3SM-Project/e3sm_diags/blob/434ab8cdd0aa0781e0064dafb0309695e714960d/e3sm_diags/derivations/acme.py#L488-L490